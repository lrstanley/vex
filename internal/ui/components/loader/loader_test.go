// Copyright (c) Liam Stanley <liam@liam.sh>. All rights reserved. Use of
// this source code is governed by the MIT license that can be found in
// the LICENSE file.

package loader

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/lrstanley/vex/internal/ui/testui"
)

func TestMain(m *testing.M) {
	v := m.Run()
	snaps.Clean(m, snaps.CleanOpts{Sort: true})
	os.Exit(v)
}

func TestNew(t *testing.T) {
	t.Parallel()
	t.Run("basic-loader", func(t *testing.T) {
		t.Parallel()
		m := New()
		m.SetHeight(testui.DefaultTermHeight)
		m.SetWidth(testui.DefaultTermWidth)
		tm := testui.NewNonRootModel(t, m, false)
		tm.ExpectViewContains(t, "loading")
		tm.ExpectViewDimensions(t, m.GetWidth(), m.GetHeight())
		tm.ExpectViewSnapshot(t)
	})

	t.Run("0-width-height", func(t *testing.T) {
		t.Parallel()
		m := New()
		m.SetHeight(0)
		m.SetWidth(0)
		tm := testui.NewNonRootModel(t, m, false)
		tm.ExpectViewSnapshot(t)
	})

	t.Run("small-dimensions", func(t *testing.T) {
		t.Parallel()
		m := New()
		m.SetHeight(5)
		m.SetWidth(20)
		tm := testui.NewNonRootModel(t, m, false)
		tm.ExpectViewContains(t, "loading")
		tm.ExpectViewSnapshot(t)
	})
}
