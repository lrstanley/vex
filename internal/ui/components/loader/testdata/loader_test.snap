
[TestNew/0-width-height - 1]

---

[TestNew/basic-loader - 1]
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                 <spinner> loading...                                   
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
---

[TestNew/small-dimensions - 1]
                    
                    
   <spinner> loading...     
                    
                    
---
